paddle/operators/math/matmul.h

```diff
@@ -41,10 +41,23 @@ class MatMulFunctor {
                       "Input tensor a must be at least 1-dimensional.");
     PADDLE_ENFORCE_GE(dim_b.size(), 1,
                       "Input tensor b must be at least 1-dimensional.");
-    PADDLE_ENFORCE_LE(dim_a.size(), 3,
-                      "Input tensor a must be at most 3-dimensional.");
-    PADDLE_ENFORCE_LE(dim_b.size(), 3,
-                      "Input tensor b must be at most 3-dimensional.");
+
+    std::vector<int64_t> out_dim;
+    int64_t batch_count = 1;
+    if (dim_a.size() > 3) {
+      PADDLE_ENFORCE(dim_b.size() > 3,
+                     "The dimensions of X and Y must be the same, and both of "
+                     "them should be %d-dimensional.",
+                     dim_b.size());
```

Comment:
It seems that the check in line48 is not consistent with its comments. It seems that the check should be like this:

```cpp
PADDLE_ENFORCE_EQ(dim_a.size(), dim_b.size(),
                  "The dimensions of X and Y must be the same."
                  "X_dim vs. Y_dim = %d vs. %d",
                  dim_a.size(), dim_b.size());
```